src/common/src/main/scala/osmesa/common/model/Change.scala

```diff
@@ -25,55 +28,64 @@ case class Change(id: Long,
                   sequence: Int)
 
 object Change {
+  def empty: Change = Change(-1, "", Map.empty, None, None, None, None, -1, "1970-01-01 00:00:00".asInstanceOf[Timestamp], -1, "", -1, false, -1)
+
   implicit def stringToTimestamp(s: String): Timestamp =
     Timestamp.from(DateTime.parse(s).toDate.toInstant)
 
-  def fromXML(node: Node, action: Action, sequence: Int): Change = {
-    val `type` = node.label
-    val id = (node \@ "id").toLong
-    val tags =
-      (node \ "tag").map(tag => (tag \@ "k", tag \@ "v")).toMap
-    val lat = node \@ "lat" match {
-      case "" => None
-      case v  => Some(v.toDouble)
-    }
-    val lon = node \@ "lon" match {
-      case "" => None
-      case v  => Some(v.toDouble)
-    }
-    val nds = `type` match {
-      case "way" =>
-        Some((node \ "nd").map(Nd.fromXML))
-      case _ => None
+  class ChangeHandler(sequence: Int) extends DefaultHandler {
+    val changeSeq = Queue.empty[Change]
```

Comment:
Why a `Queue`?

Reply:
The thought was that I wanted a fast way to update a mutable list in elementwise fashion that preserved order of insertion.  The menu of mutable collections for this kind of operation was a bit thin, so I opted for a Queue.